landscape/package/reporter.py

```diff
@@ -207,6 +211,51 @@ def _apt_update_timeout_expired(self, interval):
         last_update = os.stat(stamp).st_mtime
         return (last_update + interval) < time.time()
 
+    def _is_release_upgrader_running(self):
+        """Detect whether ubuntu-release-upgrader is running.
+
+        This is done by iterating the /proc tree (to avoid external
+        dependencies) and checkign the cmdline and the uid of the process.
+        The assumption is that ubuntu-release-upgrader is something that:
+            * is run by a python interpreter
+            * its first argument starts with '/tmp/ubuntu-release-upgrader-'
+            * is executed by root (effective uid == 0)"""
+        logging.debug("Checking if ubuntu-release-upgrader is running.")
+
+        for cmdline in glob.glob("/proc/*/cmdline"):
+            base = os.path.dirname(cmdline)
+            try:
+                with open(cmdline) as fd:
+                    read = fd.read()
+
+                pid = os.path.basename(os.path.dirname(cmdline))
+
+                cmdline = [f for f in filter(lambda x: x != '',
+                                             read.split("\x00"))]
+                if len(cmdline) <= 1:
+                    continue
+
+                with open(os.path.join(base, "status")) as fd:
+                    read = fd.read()
+
+                pattern = re.compile('^Uid\:(.*)$',
+                                     re.VERBOSE | re.MULTILINE)
+
+                for pattern in pattern.finditer(read):
+                    uuid = pattern.groups()[0].split("\t")[1]
+            except IOError:
+                continue
+
+            (executable, args) = (cmdline[0], cmdline[1:])
+
+            if (PYTHON_BIN in executable and
+                    any(RELEASE_UPGRADER_PATTERN in x for x in args) and
```

Comment:
same here. If we know the full path, I think startswith would be slightly faster to evaluate.